BASE/src/Microsoft.ApplicationInsights/Channel/Transmission.cs

```diff
@@ -133,6 +134,47 @@ public ICollection<ITelemetry> TelemetryItems
             get; private set;
         }
 
+        /// <summary>
+        /// Gets a value indicating whether flush is async.
+        /// </summary>
+        public bool HasFlushTask => this.flushTaskCompletionSource != null && !this.flushTaskCompletionSource.Task.IsCompleted;
+
+        /// <summary>
+        /// Gets Test to support async Flush.
+        /// </summary>
+        public Task<bool> GetFlushTask(CancellationToken cancellationToken)
+        {
+            if (this.flushTaskCompletionSource == null)
+            {
+                this.flushTaskCompletionSource = new TaskCompletionSource<bool>();
+                cancellationToken.Register(() => this.CancelFlushTask());
+            }
+
+            return this.flushTaskCompletionSource.Task;
+        }
+
+        /// <summary>
+        /// Send signal to flush async task on completion.
+        /// </summary>
+        public void CompleteFlushTask(bool success)
+        {
+            if (this.HasFlushTask)
+            {
+                this.flushTaskCompletionSource.TrySetResult(success);
```

Comment:
`this.flushTaskCompletionSource?.TrySetResult(success);` 

might work here as well?

Reply:
HasFlushTask is present to avoid setting flushTaskCompletionSource for the second time. It should be set only once.